src/NuGetGallery.Core/GitHub/GitHubUsageConfiguration.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NuGetGallery.GitHub
+{
+    public class GitHubUsageConfiguration : IGitHubUsageConfiguration
+    {
+        public GitHubUsageConfiguration(IReadOnlyList<RepositoryInformation> repositories)
+        {
+            if (repositories == null)
+            {
+                throw new ArgumentNullException(nameof(repositories) + " is null!");
+            }
+
+            NuGetPackagesGitHubDependencies = repositories.Any()
+                ? GetNuGetPackagesDependents(repositories)
+                : new Dictionary<string, NuGetPackageGitHubInformation>();
+        }
+
+        private Dictionary<string, NuGetPackageGitHubInformation> NuGetPackagesGitHubDependencies { get; }
+
+        public NuGetPackageGitHubInformation GetPackageInformation(string packageId)
+        {
+            if (null == packageId)
+            {
+                throw new ArgumentException(string.Format("{0} cannot be null!", nameof(packageId)));
+            }
+
+            if (NuGetPackagesGitHubDependencies.TryGetValue(packageId, out var value))
+            {
+                return value;
+            }
+
+            return new NuGetPackageGitHubInformation();
+        }
+
+        private static Dictionary<string, NuGetPackageGitHubInformation> GetNuGetPackagesDependents(IReadOnlyList<RepositoryInformation> repositories)
+        {
+            var dependentsPerPackage = new Dictionary<string, List<RepositoryInformation>>(StringComparer.InvariantCultureIgnoreCase);
+            foreach (var repo in repositories)
+            {
+                foreach (var dependency in repo.Dependencies)
+                {
+                    if (!dependentsPerPackage.TryGetValue(dependency, out var packageDependents))
+                    {
+                        packageDependents = new List<RepositoryInformation>();
+                        dependentsPerPackage[dependency] = packageDependents;
+                    }
+
+                    packageDependents.Add(repo);
+                }
+            }
+
+            return dependentsPerPackage
+                .ToDictionary(
+                    entry => entry.Key,
+                    entry =>
+                    {
+                        entry.Value.Sort(Comparer<RepositoryInformation>.Create((x, y) =>
```

Comment:
You should be able to do this with `OrderByDescending` and then `ThenBy` on `Stars` then by `Id`